File: Java/src/First Missing Positive.java
public class Solution
{
  public int firstMissingPositive(int[] A)
  {
    int length = A.length;
    for (int i = 0; i < length; i++)
    {
      int target = A[i];
      if (target > 0 && target <= length && target != A[target - 1])
      {
        int temp = A[target - 1];
        A[target - 1] = target;
        A[i] = temp;
        i--;

      }
    }
    for (int i = 0; i < length; i++)
    {
      if (A[i] != i + 1)
        return i + 1;
    }

    return length + 1;

  }
}

File: Python/First Missing Positive.py
class Solution:
    # @param A, a list of integers
    # @return an integer
    def firstMissingPositive(self, A):
        for i in range(len(A)):
            target = A[i]
            while 0 < target  and target <= len(A) and target!=A[target-1]:
                temp = A[target-1]
                A[target-1] = target
                target = temp
                
        for i in range(len(A)):
            if A[i] != i+1:
                return i+1
        return len(A)+1
        